taskcli/cmd/login.go

```diff
@@ -0,0 +1,80 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/MrShanks/Taska/utils"
+	"github.com/spf13/cobra"
+)
+
+var token string
+
+var loginCmd = &cobra.Command{
+	Use:   "login -e [email] -p [password]",
+	Short: "Login with email and password",
+	Long:  "Login with email and password to get a token to auhtenticate future requests",
+
+	Run: func(cmd *cobra.Command, args []string) {
+		apiClient := NewApiClient()
+		ctx := context.Background()
+
+		email, err := cmd.Flags().GetString("email")
+		if err != nil {
+			cmd.Printf("A email must be provided to login: %v\n", err)
+		}
+
+		password, err := cmd.Flags().GetString("password")
+		if err != nil {
+			cmd.Printf("Empty password is not allowed: %v\n", err)
+		}
+
+		if err := Login(apiClient, ctx, "signin", email, password); err != nil {
+			cmd.Printf("Login failed: %v\n", err)
+		}
+	},
+}
+
+func Login(taskcli *Taskcli, ctx context.Context, endpoint, email, password string) error {
+	taskcli.ServerURL.Path = endpoint
+
+	data := map[string]string{"email": email, "password": password}
+
+	response, err := makeRequest(taskcli, ctx, data)
```

Comment:
Since the data isn't used anywhere else, wouldn't it be cleaner to move it directly into the called function?
```suggestion

	response, err := makeRequest(taskcli, ctx, map[string]string{"email": email, "password": password})
```

Reply:
to me that makes it less readable, by giving that bloat of text a name the functions reads better.